swarm/src/connection.rs

```diff
@@ -470,24 +472,23 @@ impl<'a> IncomingInfo<'a> {
     }
 }
 
-struct SubstreamUpgrade<UserData, Upgrade> {
+struct SubstreamUpgrade<UserData, TOk, TErr> {
```

Comment:
```suggestion
struct StreamUpgrade<UserData, TOk, TErr> {
```

Given that this is an internal type, why not already rename it here?

Reply:
It didn't even occur to me, yes makes a lot of sense :)

Reply:
There are a fair few more usages of "substream" but I think those can easily go in separate PRs.

Reply:
> There are a fair few more usages of "substream" but I think those can easily go in separate PRs.

:+1: works for me.